artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/bridge/AMQPBridgeConfiguration.java

```diff
@@ -0,0 +1,317 @@
+import static org.apache.activemq.artemis.protocol.amqp.connect.bridge.AMQPBridgeConstants.DEFAULT_LINK_RECOVERY_DELAY;
+import static org.apache.activemq.artemis.protocol.amqp.connect.bridge.AMQPBridgeConstants.DEFAULT_LINK_RECOVERY_INITIAL_DELAY;
+import static org.apache.activemq.artemis.protocol.amqp.connect.bridge.AMQPBridgeConstants.DEFAULT_MAX_LINK_RECOVERY_ATTEMPTS;
+import static org.apache.activemq.artemis.protocol.amqp.connect.bridge.AMQPBridgeConstants.DEFAULT_DISABLE_RECEIVER_PRIORITY;
+import static org.apache.activemq.artemis.protocol.amqp.connect.bridge.AMQPBridgeConstants.DEFAULT_PULL_CREDIT_BATCH_SIZE;
+import static org.apache.activemq.artemis.protocol.amqp.connect.bridge.AMQPBridgeConstants.DEFAULT_QUEUE_RECEIVER_IDLE_TIMEOUT;
+import static org.apache.activemq.artemis.protocol.amqp.connect.bridge.AMQPBridgeConstants.DEFAULT_RECEIVER_QUIESCE_TIMEOUT;
+import static org.apache.activemq.artemis.protocol.amqp.connect.bridge.AMQPBridgeConstants.DEFAULT_SEND_SETTLED;
+import static org.apache.activemq.artemis.protocol.amqp.connect.bridge.AMQPBridgeConstants.DISABLE_RECEIVER_DEMAND_TRACKING;
+import static org.apache.activemq.artemis.protocol.amqp.connect.bridge.AMQPBridgeConstants.IGNORE_QUEUE_CONSUMER_FILTERS;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.activemq.artemis.protocol.amqp.proton.AMQPConnectionContext;
+import org.apache.activemq.artemis.protocol.amqp.proton.AmqpSupport;
+
+/**
+ * A configuration class that contains API for getting AMQP bridge specific configuration
+ * either from a {@link Map} of configuration elements or from the connection associated
+ * with the bridge instance, or possibly from a set default value.
+ */
+public class AMQPBridgeConfiguration {
+
+   private final Map<String, Object> properties;
+   private final AMQPConnectionContext connection;
+
+   public AMQPBridgeConfiguration(AMQPConnectionContext connection, Map<String, Object> properties) {
+      Objects.requireNonNull(connection, "Connection provided cannot be null");
+
+      this.connection = connection;
+
+      if (properties != null && !properties.isEmpty()) {
+         this.properties = new HashMap<>(properties);
+      } else {
+         this.properties = Collections.emptyMap();
+      }
+   }
+
+   /**
+    * {@return the credit batch size offered to a receiver link.}
+    */
+   public int getReceiverCredits() {
+      final Object property = properties.get(RECEIVER_CREDITS);
+      if (property instanceof Number) {
+         return ((Number) property).intValue();
+      } else if (property instanceof String) {
+         return Integer.parseInt((String) property);
+      } else {
+         return connection.getAmqpCredits();
+      }
+   }
+
+   /**
+    * {@return the number of remaining credits on a receiver before the batch is replenished.}
```

Comment:
I'm not certain whether the processor handles the trailing '.' here inside the tag, but as the typical usage of the new "{@return some detail} Additional detail" tag syntax definitely omits a '.' outside the "{}" tag, I would not be surprised if this caused it to finish up with ".." at the end of the generated "Returns foo.." text.

EDIT: it does, so need to remove the '.' inside the tag, here and various other places above/below.